src/Symfony/Component/Cache/Traits/RedisTrait.php

```diff
@@ -159,13 +159,17 @@ public static function createConnection($dsn, array $options = [])
             throw new InvalidArgumentException(sprintf('Invalid Redis DSN: "%s".', $dsn));
         }
 
-        if (isset($params['redis_sentinel']) && !class_exists(\Predis\Client::class)) {
-            throw new CacheException(sprintf('Redis Sentinel support requires the "predis/predis" package: "%s".', $dsn));
+        $params += $query + $options + self::$defaultConnectionOptions;
+
+        if (isset($params['redis_sentinel']) && (!class_exists(\Predis\Client::class) || !class_exists(\RedisSentinel::class))) {
```

Comment:
@nicolas-grekas I was doing a bit more testing on it and I might have screwed it here.

I believe it should be:
```php
if (isset($params['redis_sentinel']) && !class_exists(\Predis\Client::class) && !class_exists(\RedisSentinel::class)) {
```

I should fail if both classes are missing.

Reply:
see 620ee34b90

Reply:
Phew! Thank you! 😅